framework/tests/all-systems/BigBinaryTrees.java

```diff
@@ -0,0 +1,271 @@
+// Test to check for common problem with BinaryTrees.
+// See comment on TreeAnnotator#visitBinary.
```

Comment:
Isn't the problem mostly a timing issue, i.e. the checker will run very slowly on this example?
Or might checkers fail with a stackoverflow or some other exception?
A note would help to see whether any manual action/check is necessary.